If the user moves a file in the file system while the filter is active, the filter needs to refresh so the file disappears from the list

The application tracks files that are added, removed from, or moved in the folders that the user is working on. One bug is that if the user has a filter in place in the application and they move a file in the operating system the filter does not refresh the view to indicate that the file no longer passes the filter. Instead it shows an icon (but with no preview, because the cached preview was also moved to a different location on disk). Current incorrect behavior is that the filtered view still shows an icon for the image when it no longer passes the filter. Desired behavior is that the view should refresh to remove the item(s) that no longer pass the filter.

Problem in building this one: 
```
compendiacore.cpp:2110: error: 'void QSortFilterProxyModel::invalidateFilter()' is protected within this context
compendiacore.cpp: In member function 'void CompendiaCore::processWatcherChanges()':
compendiacore.cpp:2110:46: error: 'void QSortFilterProxyModel::invalidateFilter()' is protected within this context
 2110 |         tagged_files_proxy_->invalidateFilter();
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
```

--- a/compendiacore.cpp
+++ b/compendiacore.cpp
@@ -2107,7 +2107,7 @@ void CompendiaCore::processWatcherChanges()
     // would never re-run filterAcceptsRow() and the item would stay visible despite no
     // longer passing the folder filter.
     if (hadMoves)
-        tagged_files_proxy_->invalidateFilter();
+        tagged_files_proxy_->refreshFilter();
 }
 
 /*! \brief Handles a file that disappeared with no matching appearance elsewhere in the watched tree. */

--- a/filterproxymodel.h
+++ b/filterproxymodel.h
@@ -150,6 +150,14 @@ public:
      */
     bool isIsolated() const;
 
+    /*! \brief Forces the proxy to re-evaluate filterAcceptsRow() for every row.
+     *
+     * Exposes the protected QSortFilterProxyModel::invalidateFilter() so that
+     * external code (e.g. CompendiaCore) can trigger a full filter refresh after
+     * in-place model mutations that do not emit dataChanged.
+     */
+    void refreshFilter() { invalidateFilter(); }
+
     /*! \brief Returns the number of files in the current isolation set.
      *
      * \return Size of isolation_set_, or 0 when not isolated.

